test/Datadog.Trace.Tests/Agent/SpanFactory.cs

```diff
@@ -0,0 +1,29 @@
+using Datadog.Trace.Tagging;
+using Moq;
+
+namespace Datadog.Trace.Tests.Agent
+{
+    public static class SpanFactory
```

Comment:
Nice refactoring 👍 

One comment: I prefer naming such stuff like `SpanObjectMother` (see: https://martinfowler.com/bliki/ObjectMother.html). Why?
1. Maybe we would like to have a `SpanFactory` class in "production code"  in future
2. It makes it obvious that it is only for test code

Reply:
Seems an idea, not sure what is the current convention. If we adapt it, then we should renew the conventions -- if there are any written?

Reply:
Check it please